src/utils/makeReactNativeConfig.js

```diff
@@ -237,6 +265,74 @@ function makeReactNativeConfig(
   return config;
 }
 
+/**
+ * Creates an array of configs based on changing `env` for every
+ * platform and returns
+ */
+function makeReactNativeConfig(
+  userWebpackConfig: WebpackConfigFactory,
+  options: ConfigOptions,
+  platform: string,
+  logger: Logger = loggerUtil
+): WebpackConfig {
+  /**
+   * We should support also the old format of config
+   * 
+   * module.exports = {
+   *   entry: './index.js',
+   * };
+   * 
+   * module.exports = ({ platform }) => ({
+   *   entry: `./index.${platform}.js`,
+   * });
+   */
+  const isLegacy =
+    typeof userWebpackConfig === 'function' || !userWebpackConfig.webpack;
+
+  if (isLegacy) {
+    logger.warn(
+      'You using a deprecated style of the configuration. Please follow the docs for the upgrade.'
+    );
+
+    return DEPRECATEDMakeReactNativeConfig(
+      userWebpackConfig,
+      options,
+      platform
+    );
+  }
+
+  const { root, dev, minify, bundle, port } = options;
+
+  const env = {
+    root,
+    dev,
+    minify,
+    platform,
+    bundle,
+    port,
+  };
+
+  const {
+    webpack: webpackConfigFactory /* , ...haulConfig */,
+  } = userWebpackConfig;
+
+  if (
+    typeof webpackConfigFactory !== 'function' &&
+    typeof webpackConfigFactory !== 'object'
+  ) {
+    throw new Error(
+      'The webpack configuration must be an object or a function returning an object.'
```

Comment:
Can we add a link to the docs?

Reply:
Yeah, all documentation is still TODO, check the header of this PR.

Reply:
Updated.